analyzers/tests/SonarAnalyzer.UnitTest/TestCases/EqualityOnFloatingPoint.CSharp11.cs

```diff
@@ -4,6 +4,24 @@
 
 public class EqualityOnFloatingPoint
 {
+    void TestHalfNaN()
+    {
+        bool b;
+        var h = Half.NaN;
```

Comment:
In this case, it's not clear to me that it raises because one of them is NaN, or both, or one is tracked as NaN and compared to the other. 

Let's remove `h` and add `Half arg` method parameter to express that any unknown value compared to the constant will trigger the rule.

Same everywhere

Reply:
It was meant to also check for compliance of `h = Half.NaN`, which is not a comparison but an assignment.

Made into a method parameter and added the `// Compliant`.